### PR TITLE
CMake cleanup

### PR DESCRIPTION
**Pull request by Orphis**
This again improves the Windows compatibility.

### CODE REVIEW

File: Source/Core/Common/Common.h [hunk 1]
```diff
@@ -39,9 +39,13 @@ extern const std::string scm_distributor_str;
 
 // Since they are always around on Windows
```
Comment (ligfx) [inline Source/Core/Common/Common.h]: It'd be better to stick these in the vcxproj files and remove them from here

File: Source/Core/Common/Common.h [hunk 2]
```diff
@@ -39,9 +39,13 @@ extern const std::string scm_distributor_str;
...
 #define HAVE_WX 1
+#ifndef HAVE_OPENAL
```
Comment (ligfx) [inline Source/Core/Common/Common.h]: It'd be better to stick these in the vcxproj files and remove them from here

**Comment by ligfx** [comment]
Comment+build failures, otherwise looks good to me!

File: Source/Core/Common/Common.h (hunk 2 above)
Comment (Orphis) [inline Source/Core/Common/Common.h]: I believe the less logic or defines in the build system, the better. Also, it's quite annoying to edit it 😢

**Comment by Orphis** [comment]
This apparently breaks the headless client on Linux. Looking into it.

**Comment by Orphis** [comment]
And the headless client wasn't broken, the cli client with rendering was. It's fixed now after putting the includes in the right place (and removing useless ones).

File: Source/VSProps/Base.props [hunk 3]
```diff
@@ -55,6 +55,8 @@
       <PreprocessorDefinitions>PSAPI_VERSION=1;_M_X86=1;%(PreprocessorDefinitions)</PreprocessorDefinitions>
       <PreprocessorDefinitions>SFML_STATIC;%(PreprocessorDefinitions)</PreprocessorDefinitions>
       <PreprocessorDefinitions>CURL_STATICLIB;%(PreprocessorDefinitions)</PreprocessorDefinitions>
+      <PreprocessorDefinitions>HAVE_OPENAL=1;%(PreprocessorDefinitions)</PreprocessorDefinitions>
+      <PreprocessorDefinitions>HAVE_PORTAUDIO=1;%(PreprocessorDefinitions)</PreprocessorDefinitions>
```
Comment (ligfx) [inline Source/VSProps/Base.props]: 👍

Comment (Orphis) [inline Source/VSProps/Base.props]: Yes, it wasn't as bad as I thought, didn't have to edit tons of project files, there were the props files :)

**Pull request merged**